sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala

```diff
@@ -41,6 +41,11 @@ private object H2Dialect extends JdbcDialect {
         case _ => super.visitSQLFunction(funcName, inputs)
       }
     }
+
+    override def visitOverlay(inputs: Array[String]): String = {
+      val functionInfo = super.visitOverlay(inputs)
+      throw QueryCompilationErrors.noSuchFunctionError("H2", functionInfo)
```

Comment:
For the functions that are not supported by every DBMS, e.g. `overlay` and `width_bucket`, I think the default behavior should be not pushing down and then we enable push down in the dialect if the database supports it. Otherwise, we will need to turn off push down in all the database that don't support this function. Without doing this, we will have inconsistent behaviors before/after this PR.
For example, `SELECT * FROM ... WHERE OVERLAY ...` will fail for `MySQL` after this PR. I think before this PR it should pass.

Reply:
I'm agree not pushing down `overlay` in default, because most databases do not support it.
But the database support or not support `width_bucket`, each accounting for half.

Reply:
well, it's safer to not push down by default. BTW is `overlay` really a SQL standard function?

Reply:
Yes. `overlay` is a SQL standard function.

Reply:
OK then it's a bit tricky to decide which functions should be pushed down by default and which should not. The current policy is pushing down SQL standard functions by default, which seems reasonable. @huaxingao do we have better ideas?

Reply:
I don't have any good ideas for this. 
I guess if we want to stick to the current policy of pushing down SQL standard functions by default, then we probably need to turn off pushing  down a specific function in all the databases that don't support it. For example, in addition to turn off pushing down  `overlay` in H2, we probably should also turn off push down `overlay` in MySQL, TeraData, SqlServer. But if anybody happens to use `overlay` in SQLite, this still seems to be a breaking change to me because the `overlay` function used to work doesn't work any more.
Probably the policy should be by default, we push down the SQL standard functions that are supported by all the major databases?

Reply:
Maybe we can introduce a new property to control pushing down ANSI functions. The default is false. In this way, we won't break any of the existing applications. When sets to true, Spark pushes down all the ANSI functions and the users are responsible for turning off pushing down a specific ANSI function if their database doesn't support it.

Reply:
It seems the new property is a global switch. If so, other  ANSI function will not be pushed down together with unsupported ANSI function.

Reply:
How about this: we add a new API in JDBC dialect to decide which function to push down, which by default supports no functions. In the builtin dialects, we can add support for certain functions that are supported for sure (with tests to verify)

Reply:
Sounds good to me.